src/index.ts

```diff
@@ -414,3 +414,47 @@ export function isType<T extends AnyFunction | AnyClass>(payload: any, type: T):
   const name: string | undefined | null = (type as any).name
   return getType(payload) === name || Boolean(payload && payload.constructor === type)
 }
+
+type GlobalClassName = {
+  [K in keyof typeof globalThis]: (typeof globalThis)[K] extends AnyClass ? K : never
+}[keyof typeof globalThis]
+
+/**
+ * Checks if a value is an instance of a class or a class name. Useful when you
+ * want to check if a value is an instance of a class that may not be defined in
+ * the current scope. For example, if you want to check if a value is an
+ * `OffscreenCanvas` instance, you might not want to do the song and dance of
+ * using `typeof OffscreenCanvas !== 'undefined'` and then shimming
+ * `OffscreenCanvas` if the types aren't around.
+ *
+ * @example
+ * if (isInstanceOf(value, 'OffscreenCanvas')) {
+ *  // value is an OffscreenCanvas
+ * }
+ *
+ * @param value The value to recursively check
+ * @param class_ A string or class that the value should be an instance of
+ */
```

Comment:
@jcbhmr it's really annoying but I believe you need to copy paste the entire JSDoc onto every function overload.

Otherwise, if you hover over usage of `isInstanceOf` that matches a different overload than the top one, it won't show the JSDocs in VSCode (at least that's how it was a year ago)

Reply:
![image](https://github.com/mesqueeb/is-what/assets/61068799/ef4f8707-5b53-46ce-ac6e-23890ecc6b6d)
![image](https://github.com/mesqueeb/is-what/assets/61068799/245f5581-bb32-49f3-861c-b7cf9055ad72)
[playground](https://www.typescriptlang.org/play?#code/PQKhCgAIUgJBTANog9pAkpAhgW0gExQGMBXHeAOwBcsqBLFCgQihGHADMSKj7HIOACiwAuSAGcqAJzoUA5gEoxkmfIDcnbrwYUBwsRTIAjeFKWRDOE1I1cefXUNETpsuZAA+F46fMq3nt5WppAA3lCQUvBUJFK6WBoAvuDgRIziKIjwAHSocoJCAERYhQoKqemZOXkFggCMZUA)